skyportal/models/public_pages/public_source_page.py

```diff
@@ -0,0 +1,160 @@
+__all__ = ['PublicSourcePage']
+
+import json
+import jinja2
+import sqlalchemy as sa
+from sqlalchemy import event
+from sqlalchemy.dialects.postgresql import JSONB
+from sqlalchemy.orm import deferred
+from baselayer.app.env import load_env
+from baselayer.app.json_util import to_json
+from baselayer.app.models import (
+    Base,
+    UserAccessControl,
+    CustomUserAccessControl,
+)
+from ..source import Source
+from ..group import GroupUser
+
+from ...utils.cache import Cache, dict_to_bytes
+
+env, cfg = load_env()
+
+cache_dir = "cache/public_pages/sources"
+cache = Cache(
+    cache_dir=cache_dir,
+    max_age=cfg["misc.minutes_to_keep_public_source_pages_cache"] * 60,
+)
+
+
+def published_source_access_logic(cls, user_or_token):
+    """Return a query that filters PublicSourcePage instances based on user access."""
+    # if the user is a system admin, he can update and delete all published sources
+    # otherwise, he can only delete and update the published sources associate with his group
+    user_id = UserAccessControl.user_id_from_user_or_token(user_or_token)
+    query = sa.select(cls)
+    if not user_or_token.is_system_admin:
+        query = query.join(Source, cls.source_id == Source.obj_id)
+        query = query.join(GroupUser, Source.group_id == GroupUser.group_id)
+        query = query.filter(GroupUser.user_id == user_id)
+    return query
+
+
+class PublicSourcePage(Base):
+    """Public page of a source on a given date."""
+
+    update = delete = CustomUserAccessControl(published_source_access_logic)
+
+    id = sa.Column(sa.Integer, primary_key=True)
+
+    source_id = sa.Column(sa.String, nullable=False, doc="ID of the source")
+
+    created_at_iso = sa.Column(
+        sa.String,
+        nullable=True,
+        doc="Creation date in ISO format",
+    )
+
+    data = deferred(
+        sa.Column(JSONB, nullable=False, doc="Source data accessible on the page")
+    )
+
+    is_public = sa.Column(
```

Comment:
Yeah you right this option is only a leftover from the old implementation of the public page before the versions were added. Indeed now it is no longer useful. I can implement it but in the current versioning logic I don't see in which case it could be useful. Do you think this is useful ? @Theodlz  @mcoughlin

Reply:
Not at all you can remove that column (from both the model and the migration script btw), but also as I mentioned we already have that column called `created_at` on every table (comes from the Base class that all models heritage from), so my comment was just that this is a duplicate of something that is already there.